include/cuco/static_map.cuh

```diff
@@ -266,6 +290,33 @@ class static_map {
                  KeyEqual key_equal  = KeyEqual{},
                  cudaStream_t stream = 0);
 
+  /**
+   * @brief Erases keys in the range `[first, last)`.
+   *
+   * This function synchronizes `stream`.
+   *
+   * @tparam InputIt Device accessible input iterator whose `value_type` is
+   * convertible to the map's `value_type`
+   * @tparam Hash Unary callable type
+   * @tparam KeyEqual Binary callable type
+   * @param first Beginning of the sequence of keys
+   * @param last End of the sequence of keys
+   * @param hash The unary function to apply to hash each key
+   * @param key_equal The binary function to compare two keys for equality
+   * @param stream Stream used for executing the kernels
+   *
+   * @throw std::runtime_error if a unique erased key sentinel value was not
+   * provided at construction
+   */
+  template <typename InputIt,
+            typename Hash     = cuco::detail::MurmurHash3_32<key_type>,
+            typename KeyEqual = thrust::equal_to<key_type>>
+  void erase(InputIt first,
```

Comment:
Can we easily return the count of keys successfully erased?

Reply:
Yeah, that wouldn't be hard. Would we want to do this for insert as well?

Reply:
I thought `insert` already did, but I guess not. I think @sleeepyjack changed that such that `insert` wouldn't need to synchronize the stream.

Nevermind then, don't bother making the change to `erase`.

Reply:
For reference: #102